yt/frontends/gadget/fields.py

```diff
@@ -122,6 +128,22 @@ def _h_density(field, data):
 
         return ["H"] + metal_names[:-1]
 
+    def _check_whitelist_species_fields(self, ptype):
+        species_names = []
+        species_nums = []
+        for field in self.ds.field_list:
+            if field[0] == ptype:
+                if field[1].endswith("_fraction") or field[1].endswith("_density"):
```

Comment:
little known api convenience: `str.endswith` accepts a tuple, which does what you need here
```suggestion
                if field[1].endswith(("_fraction", "_density")):
```

Reply:
handled